### DISCUSSION TITLE
Include the updated version of Terminal in the Windows ISO

### DISCUSSION THREAD

**Discussion by ghost** [microsoft/terminal#17726] [Ideas]
I just downloaded the Windows 11 24H2 (26100.1150) ISO from the Release Preview Channel (the last testing channel before the public release).

The ISO file came with Terminal v1.18.10301.0 (January 31, 2024) installed. The latest stable version is v1.20.11781.0 (June 27, 2024).
I believe this is an old problem with Windows installations. The installation comes with the latest monthly updates, but the version of the installed applications is always old.

**Saying that the Terminal will be automatically updated to the latest version by the Store is not always true:**

1. Many people uninstall the Windows Store, because they don't like it and prefer to download EXE setups directly.
2. Many people (like me) install Windows with local accounts, this is done by turning off the internet ("oobe\bypassnro"). So the Terminal is not updated during the installation.
3. Companies, developers and even some power users disable automatic Store updates.

**My idea:**

Considering that today is August 16th, and the final 24H2 release will be released in September or October, I believe that the latest stable version of the terminal should be included by default in the ISO. Newer versions benefit from new features and bug fixes.

I noticed that v1.18.10301.0 is the latest version of 1.18. If the intention here is stability (include in the ISO a major version that will no longer receive minor updates), it could be updated to v1.19.11213.0.

**Comment by DHowett** [comment]
Thanks for the idea (and for the ping! In my case, they all go to the same place 🙂).

My engineering team feels this pain very acutely, trust me. The RTM media for 26100 was locked back in April or earlier.

There are a few things conspiring against us here, and they aren't likely to budge any time soon.

First, the schedule for RTM finalization compared to its actual release is pretty punishing for an application that updates as often as ours. Even considering that the media gets updated for the zero-day (and more) patch.

That wouldn't be quite so bad if not for the second: _MSIX packages cannot be serviced in an LCU._ Once RTM is declared, anything that isn't serviceable is sealed. This is the subject of active work between the servicing team and the app platform team.

Even though they're updating the media they do so by applying LCU packages.

This applies to all the inbox apps. It's just that not all of them are open-source, or update as often.

As a company we have gotten around this in the past by releasing supplemental media updates. Effectively, they're ISOs containing app preinstallation kits with embedded dependencies and licenses.

I'm not saying we don't want to do it, just that the platform is holding us back here.

---

On the flipside, a few things lighten the burden:

- Terminal is available on WinGet for folks who don't want to use the store
- Inbox applications don't require Microsoft accounts to update
- Enterprises (and enterprising individuals) can download the preinstallation kit from our releases page to preprovision prior to imaging

It's not great, but it's the state of the art here in 2024. ☹️
